File: HappyPuppy/C#/Boundaries.cs

public delegate void DisplayResponse(string s);
public delegate void NotificationResponse();

public interface Request
{
    string Heading { get; }
}

public interface ActionRequest: Request
{
    void Execute(DisplayResponse port);
}

public interface AdoptPuppyRequest: Request
{
    void Execute(string puppyName, string ownerName, NotificationResponse output);
}

File: HappyPuppy/Java/Boundaries.java

public class Boundaries
{
    public interface DisplayResponse
    {
        void Accept(String s);
    }

    public interface NotificationResponse
    {
        void ActionWasCompleted();
    }

    public interface Request
    {
        String Heading();
    }

    public interface ActionRequest extends Request
    {
        void Execute(DisplayResponse port);
    }

    public interface AdoptPuppyRequest extends Request
    {
        void Execute(String puppyName, String ownerName, 
                Boundaries.NotificationResponse output);
    }
}

